docs/development/core/server/kibana-plugin-server.coresetup.http.md

```diff
@@ -8,8 +8,9 @@
 
 ```typescript
 http: {
+        registerOnPreAuth: HttpServiceSetup['registerOnPreAuth'];
         registerAuth: HttpServiceSetup['registerAuth'];
-        registerOnRequest: HttpServiceSetup['registerOnRequest'];
+        registerOnPostAuth: HttpServiceSetup['registerOnPostAuth'];

```

Comment:
@eliperelman what do you think if we allocate basePath functionality under `basePath` namespace in httpSetup?  If you agree on this interface, I can implement for server side as a part of  https://github.com/elastic/kibana/issues/35816
```typescript
basePath: {
  get (key) {}
  set (key, value) {}
  prepend(url) {}
  remove (url) {}
}
```

Reply:
I think this makes sense. I like it.